components/menu/stories/menu.stories.js

```diff
@@ -190,12 +86,132 @@ export default {
+				},
+			],
+		},
+		{ type: "divider" },
+		{
+			idx: 2,
+			heading: "Menu header - With descriptions and icons",
+			id: "menu-heading-short-desc",
+			items: [
+				{
+					label: "Menu item with description",
+					description: "Short description",
+				},
+				{
+					label: "Selected item",
+					description: "This item is checked if single-select or multi-select mode is turned on",
+					isSelected: true,
+				},
+				{
+					label: "Selected item with icon",
+					iconName: "Cloud",
+					description: "This item is checked if single-select or multi-select mode is turned on",
+					isSelected: true,
+				},
+			],
+		},
+		{ type: "divider" },
+		{
+			idx: 3,
+			heading: "Menu header - With actions, icons, short descriptions, and values and longer header text that wraps",
+			id: "menu-heading-desc-icon-value",
+			hasActions: true,
+			items: [
+				{
+					label: "Menu item with action and a longer label that truncates if it is long enough to truncate",
+					iconName: "Cut",
+					description: "This item has a switch if multi-select mode is turned on.",
+				},
+				{
+					label: "Menu item with action",
+					iconName: "Copy",
+					description: "In multi-select mode, this item will be switched on. In single-select mode, this item will be checked.",
+					isSelected: true,
+				},
+				{
+					label: "Menu item with action and value",
+					iconName: "Paste",
+					description: "This item has a value. If multi-select mode is turned on, it also has a switch and the value can be used to label the switch.",
+					value: "⌘ C",
+				},
+				{
+					label: "Disabled menu item with action",
+					iconName: "Archive",
+					description: "Disabled menu item with description and icon",
+					isDisabled: true,
+				},
+			],
+		},
+		{
+			idx: 4,
+			heading: "Menu header - These menu items have drill-ins for a submenu",
+			id: "menu-heading-drillin",
+			items: [
+				{
+					label: "Menu item with drill-in",
+					isDrillIn: true,
+				},
+				{
+					label: "Menu item with drill-in and open submenu (not rendered)",
+					isDrillIn: true,
+					isOpen: true,
+				},
+				{
+					label: "Menu item with drill-in and value",
+					isDrillIn: true,
+					value: "Value",
+				},
+				{
+					label: "Disabled menu item with drill-in",
+					isDrillIn: true,
+					isDisabled: true,
+				}
+			],
+		},
+	],
+};
 
+/**
+ * When a menu is displayed within a tray, a submenu will replace the tray content when the parent menu item is
+ * selected. A submenu displays a back button (labeled by the title of the parent item) at the top of the tray to
+ * return the user to the previous level of the menu. The back arrow size scale used with the various menu sizes are
+ * small: 200, medium: 300, large: 400, and extra large: 500.
+ */
 export const TraySubmenu = MenuTraySubmenu.bind({});
+TraySubmenu.storyName = "Submenu in tray";
 TraySubmenu.argTypes = {
 	selectionMode: { table: { disable: true } },
 	hasDividers: { table: { disable: true } },
```

Comment:
This is a very interesting way to do this for each variant. I think this is the first time I'm seeing changes to the `argTypes` within a specific story. Huh!!